modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs

```diff
@@ -202,6 +204,66 @@ public Error OpenInExternalEditor(Script script, int line, int col)
                 case ExternalEditorId.None:
                     // Not an error. Tells the caller to fallback to the global external editor settings or the built-in editor.
                     return Error.Unavailable;
+                case ExternalEditorId.CustomEditor:
+                {
+                    string file = ProjectSettings.GlobalizePath(script.ResourcePath);
+                    var execCommand = _editorSettings.GetSetting(Settings.CustomExecPath).As<string>();
+                    var execArgs = _editorSettings.GetSetting(Settings.CustomExecPathArgs).As<string>();
+                    var args = new List<string>();
+                    var from = 0;
+                    var numChars = 0;
+                    var insideQuotes = false;
+                    var hasFileFlag = false;
+
+                    for (int i = 0; i < execArgs.Length; ++i)
+                    {
+                        if (execArgs[i] == '"' && (i > 0 || execArgs[i] != '\\'))
+                        {
+                            if (insideQuotes)
+                            {
+                                from++;
+                            }
+                            insideQuotes = !insideQuotes;
+                        }
+                        else if (execArgs[i] == ' ' && !insideQuotes || i == execArgs.Length - 1)
+                        {
+                            if (i == execArgs.Length - 1)
+                            {
+                                numChars++;
+                            }
+                            var arg = execArgs.Substr(from, numChars);
+                            args.Add(arg);
+
+                            if (arg.Contains("{file}"))
+                            {
+                                hasFileFlag = true;
+                            }
+
+                            from = i + 1;
+                            numChars = 0;
+                        }
+                        else
+                        {
+                            numChars++;
+                        }
+                    }
+
+                    if (!hasFileFlag)
+                    {
+                        args.Add(file);
+                    }
+
+                    for (int i = 0; i < args.Count; i++)
+                    {
+                        args[i] = args[i].Replace("{file}", file);
+                        args[i] = args[i].Replace("{line}", line.ToString());
```

Comment:
Nit: this can replace inside the already replaced filename, the file replacement should be last to avoid this.